src/firmware/types.rs

```diff
@@ -202,3 +210,512 @@ pub struct SnpPlatformStatus {
+    /// Reported TCB version used to derive the VCEK that signed this report.
+    pub reported_tcb: SNPTcbVersion,
+    _reserved_1: [u8; 24],
+    /// If MaskChipId is set to 0, Identifier unique to the chip.
+    /// Otherwise set to 0h.
+    #[cfg_attr(feature = "openssl", serde(with = "BigArray"))]
+    pub chip_id: [u8; 64],
+    /// CommittedTCB
+    pub committed_tcb: SNPTcbVersion,
+    /// The build number of CurrentVersion
+    pub current_build: u8,
+    /// The minor number of CurrentVersion
+    pub current_minor: u8,
+    /// The major number of CurrentVersion
+    pub current_major: u8,
+    _reserved_2: u8,
+    /// The build number of CommittedVersion
+    pub committed_build: u8,
+    /// The minor number of CommittedVersion
+    pub committed_minor: u8,
+    /// The major number of CommittedVersion
+    pub committed_major: u8,
+    _reserved_3: u8,
+    /// The CurrentTcb at the time the guest was launched or imported.
+    pub launch_tcb: SNPTcbVersion,
+    #[cfg_attr(feature = "openssl", serde(with = "BigArray"))]
+    _reserved_4: [u8; 168],
+    /// Signature of bytes 0 to 0x29F inclusive of this report.
+    /// The format of the signature is found within [`Signature`].
+    pub signature: Signature,
+}
+
+/// The response from the PSP containing the generated attestation report.
+/// According to the kernel, this should match [`self::guest::ReportRsp`].
+/// The Report is padded to exactly 4000 Bytes to make sure the page size
+/// matches.
+///
+///
+/// ```txt
+///     96 Bytes (*Message Header)
+/// + 4000 Bytes (*Encrypted Message)
+/// ------------
+///   4096 Bytes (4k Memory Page Alignment)
+/// ```
+/// <sup>*[Message Header - 8.26 SNP_GUEST_REQUEST - Table 97](<https://www.amd.com/system/files/TechDocs/56860.pdf#page=113>)</sup>
+///
+/// <sup>*[Encrypted Message - sev-guest.h](<https://git.kernel.org/pub/scm/linux/kernel/git/tip/tip.git/tree/include/uapi/linux/sev-guest.h>)</sup>
+#[repr(C)]
+pub struct MsgReportRsp {
+    /// The status of key derivation operation.
+    ///     0h: Success.
+    ///     16h: Invalid parameters.
+    pub status: u32,
+    /// Size in bytes of the report.
+    pub report_size: u32,
+    reserved_0: [u8; 24],
+    /// The attestation report generated by the firmware.
+    pub report: AttestationReport,
+    /// Padding bits to meet the memory page alignment.
+    reserved_1: [u8; 2780],
+}
+
+/// The structure used to define the read/write data for the IOCTL
+/// SNP_GUEST_REPORT.
+#[repr(C)]
+pub struct GuestMsg<'a, 'b> {
+    /// Message version number (must be non-zero)
+    pub msg_version: u8,
+
+    /// Request structure address.
+    pub req_data: &'a mut ReportReq,
+
+    /// Response structure address.
+    pub rsp_data: &'b mut MsgReportRsp,
+
+    /// Firmware error.
+    pub fw_err: u64,
+}
+
+/// Corresponds to MSG_KEY_REQ in the Spec. The message structure that the guest sends to the firmware to
+/// request a derived key.
+#[repr(C)]
+pub struct DerivedKey {
```

Comment:
maybe `DerivedKeyRequest` ? or `DeriveKeyRequest` ?

Reply:
Should we opt for `DerivedKeyRequest`, or `GetDerivedKey` to match the C Macros?